client/packages/executor/src/executionManager/sideEffect.ts

```diff
@@ -273,13 +276,16 @@ export class SideEffect extends EventEmitter {
      * used to determine if another bid should be placed.
      */
     recomputeMaxProfit() {
-        const txCostUsd = this.gateway.toFloat(this.txCostNative.getValue()) * this.nativeAssetPrice.getValue()
+        const txCostUsd = Big(this.gateway.toFloat(this.txCostNative.getValue())).times(this.nativeAssetPrice.getValue())
```

Comment:
We run into the same problem discussed before here. We can't use `toFloat()` as it breaks precision.

Reply:
AmountConverter also doesn't contain a safe version of this functionality, which was outlined in the ticket.